### PR TITLE
tau: add v2.32.1

### PR DESCRIPTION
**Pull request by eugeneswalker**
@sameershende

### CODE REVIEW

**Review by sameershende — APPROVED** [review]

**Review by kwryankrattiger — DISMISSED** [review]

**Comment by alecbcs** [comment]
@eugeneswalker looks like a legitimate failure in CI attempting to build tau.

**Comment by eugeneswalker** [comment]
> @eugeneswalker looks like a legitimate failure in CI attempting to build tau.

Yes, I am investigating locally. The package.py will need to be updated with additional logic for handling `tau@2.32.1 +rocm` before this PR is ready.

**Comment by eugeneswalker** [comment]
`tau +rocm` build is failing:
* [tau/5g5mc4j 2.32.1 gcc@=11.1.0 linux-ubuntu20.04-x86_64_v3 E4S](https://gitlab.spack.io/spack/spack/-/jobs/8009075)
```
...
In file included from Tau_plugin_monitoring.cpp:49:
tau_rocm_smi.hpp:11:10: fatal error: rocm_smi/rocm_smi.h: No such file or directory
   11 | #include "rocm_smi/rocm_smi.h"
      |          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
...
  >> 1263    tau_rocm_smi.hpp:11:10: fatal error: rocm_smi/rocm_smi.h: No such 
             file or directory
     1264       11 | #include "rocm_smi/rocm_smi.h"
     1265          |          ^~~~~~~~~~~~~~~~~~~~~
     1266    compilation terminated.
  >> 1267    make[2]: *** [Makefile:177: Tau_plugin_monitoring.o] Error 1
     1268    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-tau-
             2.32.1-5g5mc4jclpg4tipnrt76s3dgf6hfwdgt/spack-src/plugins/examples
             '
```

**Comment by khuck** [comment]
Yeah, this was a recent change to TAU to try to pull in ROCm SMI data for the monitoring plugin. For some reason the wrong include path is getting used. I need to patch TAU.

**Comment by khuck** [comment]
@eugeneswalker What's the deal with the E4S version of rocm? which version of rocm is this, and what are the values of these variables in include/Makefile?

```
TAU_ROCM_DIR=/opt/rocm-5.5.1
TAU_ROCM_SMI_INC=$(TAU_ROCM_DIR)/include/rocm_smi
TAU_ROCM_SMI_LIB_DIR=$(TAU_ROCM_DIR)/rocm_smi/lib
```

**Comment by eugeneswalker** [comment]
The `tau +rocm` issue has been resolved now via addition of `-rocmsmi=<path>` configure arg and accompanying modifications to TAU's package.py logic for `@2.32.1 +rocm`.

**Comment by eugeneswalker** [comment]
All CI is green and this is approved by maintainer. This is ready for merge!

File: var/spack/repos/builtin/packages/tau/package.py [hunk 1]
```diff
@@ -124,6 +125,7 @@ class Tau(Package):
     depends_on("rocprofiler-dev", when="+rocprofiler")
     depends_on("roctracer-dev", when="+roctracer")
     depends_on("hsa-rocr-dev", when="+rocm")
+    depends_on("rocm-smi-lib", when="+rocm")
```
Comment (becker33) [inline var/spack/repos/builtin/packages/tau/package.py]: Has this always been true (and just not noted properly) or is this new with this version? If it's the latter we need to apply it only for this version and onward.

**Review by becker33 — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/tau/package.py (hunk 1 above)
Comment (eugeneswalker) [reply var/spack/repos/builtin/packages/tau/package.py]: This only applies to `@2.32.1 +rocm` -- I updated the constraint just now and pushed.

**Review by becker33 — APPROVED** [review]

**Pull request merged**